### PR TITLE
E-828 - Allow leads to edit their project's repositories - QA 01

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: src/pages/ProjectDetails/ProjectEdition/EditContext.tsx [hunk 1]
```diff
@@ -70,33 +71,51 @@ export const EditContext = createContext<Edit>({
...
-    ignoreContributionsBefore: z.coerce.date().optional(),
-  }),
-});
-
 const SESSION_KEY = "edit-project-";
 
 export function EditProvider({ children, project }: EditContextProps) {
   const { T } = useIntl();
+
+  const validationSchema = z.object({
```
[the remark this answers is not in the document]
Comment (alexbeno) [reply src/pages/ProjectDetails/ProjectEdition/EditContext.tsx:79]: we can do an hook maybe but we need to use useIntl for error messages. What do you think ?

[the remark this answers is not in the document]
Comment (alexbeno) [reply src/pages/ProjectDetails/ProjectEdition/EditContext.tsx:79]: useValidationSchema ?

File: src/pages/ProjectDetails/ProjectEdition/pages/Information.tsx [hunk 2]
```diff
@@ -121,21 +124,33 @@ export function Information() {
...
                   key="moreInfos.url"
                   name="moreInfos.url"
-                  value={value?.[0].url}
+                  value={value?.[0]?.url}
                   fieldClassName="flex-1"
+                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                  errorMessage={(error as any)?.[0]?.url?.message}
                   onChange={event => {
-                    onChangeField([{ ...(value?.[0] || {}), url: event.target.value }]);
+                    if (!event.target.value && !value?.[0].value) {
```
[the remark this answers is not in the document]
Comment (alexbeno) [reply src/pages/ProjectDetails/ProjectEdition/pages/Information.tsx:132]: with an else if ?

File: src/pages/ProjectDetails/ProjectEdition/EditContext.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/pages/ProjectDetails/ProjectEdition/EditContext.tsx:79]: yes a hook that handle also translation for error messages can be a good idea 👍

File: src/pages/ProjectDetails/ProjectEdition/pages/Information.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/pages/ProjectDetails/ProjectEdition/pages/Information.tsx:132]: just like we did in other cases by returning in prior a non empty value, else return empty value :

`if (event.target.value && value?.[0].value) {
onChangeField([{ ...(value?.[0] || {}), url: event.target.value }]);
} else {
onChangeField([]);
}
`

**Pull request merged**